api/novelist.py

```diff
@@ -551,31 +569,50 @@ def create_item_object(self, object, currentIdentifier, existingItem):
             # a data error.
             return (None, None, None, False)
 
-        role = object[5]
-        author = object[6] if role in Contributor.AUTHOR_ROLES else ""
+        roles = list(Contributor.AUTHOR_ROLES)
+        roles.append(Contributor.NARRATOR_ROLE)
+
+        role = object[6]
+        author_or_narrator = object[7] if role in roles else ""
+        distributor = object[8]
 
         # If we encounter an existing ISBN and its role is "Primary Author",
         # then that value overrides the existing Author property.
-        #
-        # TODO: add 'narrator' field when we encounter a Narrator role.
+        # If the role is narrator, then we don't need to worry about this step.
         if isbn == currentIdentifier and existingItem:
             if role == Contributor.PRIMARY_AUTHOR_ROLE:
-                existingItem['author'] = author
+                existingItem['author'] = author_or_narrator
                 existingItem['role'] = role
             return (currentIdentifier, existingItem, None, False)
         else:
             # If we encounter a new ISBN, we take whatever author value is
             # initially given to us.
             title = object[3]
             mediaType = self.medium_to_book_format_type_values.get(object[4], "")
+
             newItem = dict(
                 isbn=isbn,
                 title=title,
                 mediaType=mediaType,
-                author=author,
                 role=role,
+                distributor=distributor
             )
-            return (isbn, existingItem, newItem, True)
+
+            publicationDate = object[5]
+            if publicationDate:
```

Comment:
This gave me an issue in a test because the value was None. There can be books without a publication date?

Reply:
Yeah, publication date is just a piece of information that the distributor is supposed to provide, and some distributors don't provide it.